src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java

```diff
@@ -292,6 +305,39 @@ public UUID getStudyRootNodeUuid(UUID studyId) {
         return nodesRepository.findByStudyIdAndType(studyId, NodeType.ROOT).orElseThrow(() -> new StudyException(ELEMENT_NOT_FOUND)).getIdNode();
     }
 
+    private boolean isNodeExist(UUID studyUuid, String nodeName) {
+        return getStudyNodesNames(studyUuid).contains(nodeName);
+    }
+
+    private List<String> getStudyNodesNames(UUID studyUuid) {
+        return nodesRepository.findAllByStudyId(studyUuid)
+                .stream()
+                .filter(nodeEntity -> nodeEntity.getType() == NodeType.NETWORK_MODIFICATION)
+                .map(nodeEntity -> networkModificationNodeInfoRepository.findById(nodeEntity.getIdNode()).get().getName())
+                .collect(Collectors.toList());
+    }
+
+    public String getUniqueNodeName(UUID studyUuid) {
+        String namePrefix = "New node ";
+        int counter = 0;
```

Comment:
Maybe you should have a static counter here, to not restart at 0 each time you create a node...